layabout.py

```
@@ -1,3 +1,287 @@
+
+        Returns:
+            Whether the connection succeeded.
+
+        Raises:
+            MissingSlackToken: If no API token is available.
+        """
+        resetting = False
+
+        # If we were given a token let's prefer the new one and establish or
+        # reset the connection.
+        if token is not None:
+            self._token = token
+            resetting = True
+
+        # We don't have an existing token, so let's try to get one from an
+        # environment variable or give up.
+        if self._token is None:
+            try:
+                self._token = os.environ[self._env_var]
+            except KeyError:
+                raise MissingSlackToken('Cannot connect to the Slack API'
+                                        ' without a token')
+
+        # Either we've never connected before or we're purposefully resetting
+        # the connection.
+        if self.slack is None or resetting:
+            # TODO: Maybe set an appropriate user agent string here using
+            # self.name.
+            self.slack = SlackClient(token=self._token)
+
+        # Use whatever token we've got to attempt to connect (or reconnect).
+        return self.slack.rtm_connect()
+
+    def _reconnect(self, retries: int,
+                   backoff: Callable[[int], float]) -> bool:
+        """
+        Attempt to reconnect to the Slack API.
+
+        Args:
+            retries: The number of retry attempts to make if a connection
+                to Slack if not established or is lost.
+            backoff: The strategy used to determine how
+                long to wait between retries. Must take as input the number of
+                the current retry and output a :obj:`float`.
+
+        Returns:
+            Whether the reconnection succeeded.
+        """
+        # TODO: Should retries start at 0 or 1?
+        for retry in range(retries):
+            if self.connect():
+                return True
```

It's not clear from reading if `_reconnect` and `connect` have the same return type or not. `connect` returns the call to `rtm_connect()` and here we return a Boolean. Does `rtm_connect()`  also return a Boolean?

---

As denoted by their type annotations, both methods have the same return type:

**connect**
```python
def connect(self, token: str = None) -> bool: ...
```
**_reconnect**
```python
def _reconnect(self, retries: int,
               backoff: Callable[[int], float]) -> bool: ...
```
The `SlackClient.rtm_connect` method eats any exceptions and always returns a `bool`.